src/transformers/generation_utils.py

```diff
@@ -712,15 +713,33 @@ def _get_logits_processor(
             processors.append(ForcedEOSTokenLogitsProcessor(max_length, forced_eos_token_id))
         if remove_invalid_values is True:
             processors.append(InfNanRemoveLogitsProcessor())
+        processors = self._merge_criteria_processor_list(processors, logits_processor)
         return processors
 
-    def _get_stopping_criteria(self, max_length: Optional[int], max_time: Optional[float]) -> StoppingCriteriaList:
-        stopping_criteria = StoppingCriteriaList()
+    def _get_stopping_criteria(
+        self, max_length: Optional[int], max_time: Optional[float], stopping_criteria: Optional[StoppingCriteriaList]
+    ) -> StoppingCriteriaList:
+        criteria = StoppingCriteriaList()
         if max_length is not None:
-            stopping_criteria.append(MaxLengthCriteria(max_length=max_length))
+            criteria.append(MaxLengthCriteria(max_length=max_length))
         if max_time is not None:
-            stopping_criteria.append(MaxTimeCriteria(max_time=max_time))
-        return stopping_criteria
+            criteria.append(MaxTimeCriteria(max_time=max_time))
+        criteria = self._merge_criteria_processor_list(criteria, stopping_criteria)
+        return criteria
+
+    def _merge_criteria_processor_list(
+        self, default_list: StoppingCriteriaList, custom_list: StoppingCriteriaList
```

Comment:
```suggestion
        self, default_list: List[Any], custom_list: List[Any]
```

Reply:
They are not  of type `List`. `Union[LogitsProcessorList, StoppingCriteriaList]` is the correct thing (you can define a new type too to make it less verbose maybe)

```python
GenerateList = Union[LogitsProcessorList, StoppingCriteriaList]

    def _merge(... default_list: GenerateList) -> GenerateList:
       ....
```

Reply:
I have added `Union[LogitsProcessorList, StoppingCriteriaList]`. Should be more careful with type hints in the future. I chose the verbose option because it is quite readable that way.